docs/rules/structured-imports.md

```diff
@@ -0,0 +1,33 @@
+# Enforces a certain structure in your import statements
+
+## Rationale
+- Consistency in import statement structure throughout the codebase
+
+## Rule Details
+Enforce a certain structure in your imports. Import structure should be listed in the following order:
+
+* Module imports
+* Absolute imports
+* Relative parent/ancestor directory imports
+* Relative sibling directory imports
+
+The following are considered warnings
+```js
+import Foo from './foo';
+import Baz = '../baz';
+import BarModule from 'BarModule';
+```
+
+The following patterns are not warnings:
```

Comment:
I'd prefer: `The following import order is valid:`

Sometimes coding rules apply to English!  Less cognitive load to interpret "valid" than "not warnings".